utils/BreakUpMomentum.py

```diff
@@ -7,7 +7,10 @@
 
 import argparse
 from ROOT import TDatabasePDG
+import math
 
+def ComputeSof2Particles(m1, k1, m2, k2):
+    return m1*m1 + m2*m2 + 2*math.sqrt(m1*m1 + k1*k1)*math.sqrt(m2*m2 + k2*k2) + 2*k1*k2
```

Comment:
The last term of the equation has a + only if you assume that $\cos(k1, k2) = -1$. If you want to keep the + I would force k1 and k2 to be positive using `abs()` and add a warning in case a negative k1 or k2 is passed